monitoring/monitorlib/kml.py

```diff
@@ -0,0 +1,101 @@
+    kml = parser.parse(kml_path)
+    return kml.getroot()
+
+
+def get_folders(root):
+    return root.Document.Folder.Folder
+
+
+def get_polygon_speed(polygon_name):
+    """Returns speed unit within a polygon."""
+    result = re.search(r"\(([0-9.]+)\)", polygon_name)
+    return float(result.group(1)) if result else None
+
+def get_folder_details(folder_elem):
+    speed_polygons = {}
+    alt_polygons = {}
+    operator_location = {}
+    coordinates = ''
+    for placemark in folder_elem.xpath('.//kml:Placemark', namespaces=KML_NAMESPACE):
+        placemark_name = str(placemark.name)
+        polygons = placemark.xpath('.//kml:Polygon', namespaces=KML_NAMESPACE)
+
+        if placemark_name == 'operator_location':
+            operator_location = {
+                'lng': folder_elem.xpath(
+                    './/kml:Placemark/kml:LookAt/kml:longitude', namespaces=KML_NAMESPACE)[0],
+                'lat': folder_elem.xpath(
+                    './/kml:Placemark/kml:LookAt/kml:latitude', namespaces=KML_NAMESPACE)[0]
+            }
+        if polygons:
+            if placemark_name.startswith('alt:'):
+                polygon_coords = get_coordinates_from_kml(
+                    polygons[0].outerBoundaryIs.LinearRing.coordinates)
+                alt_polygons.update({placemark_name: polygon_coords})
+            if placemark_name.startswith('speed:'):
+                if not get_polygon_speed(placemark_name):
+                    # TODO: raise error
+                    pass
+                polygon_coords = get_coordinates_from_kml(
+                    polygons[0].outerBoundaryIs.LinearRing.coordinates)
+                speed_polygons.update({placemark_name: polygon_coords})
+        
+        coords = placemark.xpath('.//kml:LineString/kml:coordinates', namespaces=KML_NAMESPACE)
+        if coords:
+            coordinates = coords
+            coordinates = get_coordinates_from_kml(coordinates)
+    return  {
+        str(folder_elem.name): {
+            'description': get_folder_description(folder_elem),
+            'speed_polygons': speed_polygons,
+            'alt_polygons': alt_polygons,
+            'input_coordinates': coordinates,
+            'operator_location': operator_location
+    }}
+
+
+def get_coordinates_from_kml(coordinates):
+    """Returns list of tuples of coordinates.
+    Args:
+        coordinates: coordinates element from KML.
+    """
+    if coordinates:
+       return [tuple(float(x.strip()) for x in c.split(',')) for c in str(coordinates[0]).split(' ') if c.strip()]
+
+
+def get_folder_description(folder_elem, to_json=True):
+    """Returns folder description from KML.
+    Args:
+        folder_elem: Folder element from KML.
+        to_json: If True: Returns description details as key: value pair, else a description string.
+    """
+    if to_json:
+        description = folder_elem.description
+        return dict([tuple((i.strip()).split(':')) for i in str(description).split('\n')])
```

Comment:
I think this may lead to value strings having a leading space.  Either way, I'm observing the output JSON containing strings like `" Helicopter"` rather than `"Helicopter"`.  We may want to just `trim()` value strings to get rid of leading and trailing whitespace.

Reply:
`i.strip()` was meant to remove the spaces but it required some modifications. done.